crowbar_framework/app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class RepositoriesController < ApplicationController
+  before_filter :load_repository_registry
+  #
+  # Repository Check
+  #
+  # Provides the restful api call for
+  # Repository Checks 	/utils/repositories 	GET 	Returns a json list of checked repositories
+  # Renders an HTML view in the UI
+  def index
+    @repocheck = Crowbar::Repository.check_all_repos
+    respond_to do |format|
+      format.html { @repocheck }
+      format.xml { render xml: @repocheck }
+      format.json { render json: @repocheck }
+    end
+  end
+
+  # update the state of the repositories (active/disabled)
+  def sync
+    unless params["repo"].nil?
+      provisioner_service.synchronize_repositories(params["repo"])
+    end
+
+    redirect_to repositories_path
+  end
+
+  #
+  # Activate a single Repository
+  #
+  # Provides the restful api call for
+  # Activate a Repository   /utils/repositories/activate   POST  Creates Repository DataBagItem
+  # required parameters: platform, repo
+  def activate_repository
+    if params["platform"] && params["repo"]
+      repo_object = Crowbar::Repository.where(platform: params["platform"], repo: params["repo"]).first
+      provisioner_service.create_repository_item(repo_object) unless repo_object.nil?
+    end
+  end
+
+  #
+  # Deactivate a single Repository
+  #
+  # Provides the restful api call for
+  # Deactivate a Repository   /utils/repositories/deactivate   POST   Destroys Repository DataBagItem
+  # required parameters: platform, repo
+  def deactivate_repository
+    if params["platform"] && params["repo"]
```

Comment:
Please give a proper response to an HTML (without suffix like .json) request AND to a JSON request. `head :ok` should be fine if everything went well for JSON, for the default (HTML) you should redirect to the repositories page. In a case of an error it should be like https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/app/controllers/machines_controller.rb#L95-L98